File: internal/blob/blob.go
package blob

import (
	"context"
	"io"
)

// Store is the interface for blob storage backends.
type Store interface {
	Get(ctx context.Context, userID int64, path string) (io.ReadCloser, error)
	Put(ctx context.Context, userID int64, path string, data []byte) error
	Delete(ctx context.Context, userID int64, path string) error
	DeleteTree(ctx context.Context, userID int64, folderPath string) error
	Close() error
}

File: src/Rstash.Storage/Blob.cs
namespace Rstash.Storage;

/// <summary>A blob row in the database backend, keyed by (UserId, Path).</summary>
public class Blob
{
    public long UserId { get; set; }

    public string Path { get; set; } = string.Empty;

    public byte[] Data { get; set; } = [];
}
